File: Client/Assets/Scripts/Common/Table/TableItemHeader.cs
namespace SDK.Common
{
    public class TableItemHeader
    {
        public uint m_uID;              // 唯一 ID
        public uint m_offset;           // 这一项在文件中的偏移

        // 解析头部
        virtual public void parseHeaderByteBuffer(ByteBuffer bytes)
        {
            m_uID = bytes.readUnsignedInt32();
            m_offset = bytes.readUnsignedInt32();
        }
    }
}

File: Client_Start/Assets/LuaFramework/Lua/MyLua/Libs/Table/TableItemHeader.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

local M = GlobalNS.Class(GlobalNS.GObject);
M.clsName = "TableItemHeader";
GlobalNS[M.clsName] = M;

function M:ctor()
    self.m_uID = 0;              -- 唯一 ID
    self.m_offset = 0;           -- 这一项在文件中的偏移
end

-- 解析头部
function M:parseHeaderByteBuffer(bytes)
    _, self.m_uID = bytes:readUnsignedInt32(self.m_uID);
    _, self.m_offset = bytes:readUnsignedInt32(self.m_offset);
end

return M;